[Subscriptions in Order Creation] Update Order Details, and new customer section

iamgabrielma opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #12440

On hold until the Customer bit has been implemented. Discussion: p1714034761901509-slack-C06C3CZFHGD . Additionally:

> For the mobile apps, unless we plan to fetch the subscription (shown in wp-admin Subscriptions if there’s API support) when we detect a subscription product in the order, we probably want to disable editing the addresses for sub orders because this only updates the order in Orders but not in Subscriptions

## Description
This PR addresses updating the Order Details row for billing, shipping, and customer details following the design proposal on Q9tkFMtymZz0yhfLLbKhig-fi-2176_61285 , under `i1 - 06 - Order details`

<img width="328" alt="Screenshot 2024-04-25 at 15 41 05" src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/6af3e788-86b5-45df-bc8d-45da56b41a37">

## Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->

iamgabrielma committed:
Move customer note button to end of section

iamgabrielma committed:
Update billing label. Move localization to extension

iamgabrielma committed:
Split Localizations from other extensions

iamgabrielma committed:
Update shipping details and method labels

iamgabrielma committed:
Render customer details section in order details

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

iamgabrielma committed:
Revert "Render customer details section in order details"

This reverts commit 006c6e1eae950c010390f3bd2ed1ae3a9a328484.

iamgabrielma committed:
Make Localization internal

We're using the localized titles to locate the sections in tests within `OrderDetailsDataSourceTests`. For now this needs to remain internal to buid.

iamgabrielma commented:
Closing for the moment, since the dependency project has been paused in favour of other priorities.
